common/models/templates/src/trie-model.ts

```diff
@@ -24,535 +26,525 @@
-   * Unigram models throw away all preceding words, and search
-   * for the next word exclusively. As such, they can perform simple
-   * prefix searches within words, however they are not very good
-   * at predicting the next word.
+   * Indicates that the model's written form has 'casing' behavior.
    */
-  export class TrieModel implements LexicalModel {
-    configuration: Configuration;
-    private _trie: Trie;
-    readonly breakWords: WordBreakingFunction;
-    readonly punctuation?: LexicalModelPunctuation;
-    readonly languageUsesCasing?: boolean;
-
-    readonly applyCasing?: CasingFunction;
-
-    constructor(trieData: object, options: TrieModelOptions = {}) {
-      this.languageUsesCasing = options.languageUsesCasing;
-      this.applyCasing = options.applyCasing;
-
-      this._trie = new Trie(
-        trieData['root'],
-        trieData['totalWeight'],
-        options.searchTermToKey as Wordform2Key || defaultSearchTermToKey
-      );
-      this.breakWords = options.wordBreaker || getDefaultWordBreaker();
-      this.punctuation = options.punctuation;
-    }
+  languageUsesCasing?: boolean;
 
-    configure(capabilities: Capabilities): Configuration {
-      return this.configuration = {
-        leftContextCodePoints: capabilities.maxLeftContextCodePoints,
-        rightContextCodePoints: capabilities.maxRightContextCodePoints
-      };
-    }
+  /**
+   * Specifies a function used to apply the language's casing rules to a word.
+   */
+  applyCasing?: CasingFunction;
 
-    toKey(text: USVString): USVString {
-      return this._trie.toKey(text);
-    }
+  /**
+   * Any punctuation to expose to the user.
+   */
+  punctuation?: LexicalModelPunctuation;
+}
 
-    predict(transform: Transform, context: Context): Distribution<Suggestion> {
-      // Special-case the empty buffer/transform: return the top suggestions.
-      if (!transform.insert && !context.left && !context.right && context.startOfBuffer && context.endOfBuffer) {
-        return makeDistribution(this._trie.firstN(MAX_SUGGESTIONS).map(({text, p}) => ({
-          transform: {
-            insert: text,
-            deleteLeft: 0
-          },
-          displayAs: text,
-          p: p
-        })));
-      }
+/**
+ * Used to determine the probability of an entry from the trie.
+ */
+type TextWithProbability = {
+  text: string;
+  // TODO: use negative-log scaling instead?
+  p: number; // real-number weight, from 0 to 1
+}
 
-      // Compute the results of the keystroke:
-      let newContext = models.applyTransform(transform, context);
+/**
+ * @class TrieModel
+ *
+ * Defines a trie-backed word list model, or the unigram model.
+ * Unigram models throw away all preceding words, and search
+ * for the next word exclusively. As such, they can perform simple
+ * prefix searches within words, however they are not very good
+ * at predicting the next word.
+ */
+export default class TrieModel implements LexicalModel {
+  configuration?: Configuration;
```

Comment:
I don't understand how this and a number of other changes in this file relate to ESModules?